src/run_spineopt_basic.jl

```diff
@@ -635,6 +636,123 @@ function _save_other_values!(m::Model)
     end
 end
 
+"""
+Generalized function to initialize and populate economic parameter values.
+"""
+function _initialize_and_populate_values!(
+    m::Model,
+    param_name::Symbol,
+    indices_func,
+    value_func,
+    entity::Symbol,
+    user_outputs
+)
+    
+    key_type = NamedTuple{(
+        entity, :stochastic_scenario, :t),
+        Tuple{SpineInterface.Object, SpineInterface.Object, SpineInterface.TimeSlice}
+    }
+    
+    if !use_economic_representation(model=m.ext[:spineopt].instance)
+        m.ext[:spineopt].values[param_name] = Dict{key_type,Float64}()
+        return nothing
+    end
+
+    m.ext[:spineopt].values[param_name] = 
+        Dict{key_type,Float64}(
+            (entity=>e, stochastic_scenario=s, t=t) =>
+            value_func(; entity=>e, stochastic_scenario=s, t=t)
+            for (e, s, t) in indices_func(m) if param_name in user_outputs
```

Comment:
Really? I still see the check at the end of the comprehension in master? Or I am missing something?

Reply:
Sorry, I read your first comment wrong. Yes, it is like this in the master. I now see your point, and yes, it can be improved. I will work on that and ask you to review the new PR. Thanks!